bindings/rust/wasmedge-sys/src/instance/memory.rs

```diff
@@ -248,11 +237,19 @@ impl Memory {
 }
 impl Drop for Memory {
     fn drop(&mut self) {
-        if !self.registered && !self.inner.0.is_null() {
+        if !self.registered && Arc::strong_count(&self.inner) == 1 && !self.inner.0.is_null() {
             unsafe { ffi::WasmEdge_MemoryInstanceDelete(self.inner.0) };
         }
     }
 }
+impl Clone for Memory {
+    fn clone(&self) -> Self {
+        Self {
+            inner: self.inner.clone(),
```

Comment:
I am not familiar with this one(`Arc`). But is it fine to share the pointer(C-level memory pointer) between multiple duplicated memory? Will it cause any data lock issue?

Reply:
There is no data lock issue, as `Arc` is just an atomic reference counter and has no locking mechanism. Thanks a lot!